apps/api/schedule.py

```diff
@@ -169,6 +170,69 @@ def post(self, proposal_type):
         return [t.id for t in res]
 
 
+class ProposalC3VOCPublishingWebhook(Resource):
+    method_decorators = {"post": [_require_video_api_key]}
+
+    def post(self):
+        if not request.is_json:
+            abort(415)
+
+        payload = request.get_json()
+
+        try:
+            conference = payload["fahrplan"]["conference"]
+            proposal_id = payload["fahrplan"]["id"]
+        except KeyError:
+            abort(422)
+
+        if not payload["is_master"]:
+            # c3voc *should* only send us information about the master
+            # encoding getting published. Aborting early ensures we don't
+            # accidentially delete video information from the database.
+            abort(403)
+
+        if conference != f"emf{event_year()}":
+            abort(422)
+
+        proposal = Proposal.query.get_or_404(proposal_id)
+
+        if payload["voctoweb"]["enabled"]:
+            if payload["voctoweb"]["frontend_url"]:
+                proposal.c3voc_url = payload["voctoweb"]["frontend_url"]
+                proposal.video_recording_lost = False
+            else:
+                # This allows c3voc to notify us if videos got depublished
+                # as well. We do not explicitely set 'video_recording_lost'
+                # here because the video might only need fixing audio or
+                # such.
+                proposal.c3voc_url = ""
+
+        if payload["youtube"]["enabled"]:
+            if payload["youtube"]["urls"]:
+                # Please do not overwrite existing youtube urls
+                if not proposal.youtube_url:
+                    # c3voc will send us a list, even though we only have one
+                    # video.
+                    proposal.youtube_url = payload["youtube"]["urls"][0]
+                    proposal.video_recording_lost = False
+                elif proposal.youtube_url not in payload["youtube"]["urls"]:
+                    # c3voc sent us some urls, but none of them are matching
+                    # the url we have in our database.
+                    app.logger.warning(
+                        f"C3VOC webhook sent youtube urls {payload['youtube']['urls']!r}, "
+                        f"but we already have {proposal.youtube_url}. NOT "
+                        "overwriting!"
+                    )
+            else:
+                # see comment at c3voc_url above
+                proposal.youtube_url = ""
+
+        db.session.add(proposal)
+        db.session.commit()
+
+        return "OK", 204
```

Comment:
Nitpick: we should be able to omit this (the default return status code is `200` / success; it looks like the `voctopublish` client code should handle that without a problem).

Reply:
Yeah, but since i've built some other APIs in the past, i feel only returning "OK" warrants a `204 no content` response. If we return 200 here, we should return (parts of) the proposal object, which feels unnecessary here (noone ever looks at it).

Reply:
Seems reasonable to me.  Thank you for the explanation :+1: